services/frontend-service/pkg/cmd/server.go

```diff
@@ -230,14 +230,38 @@ func runServer(ctx context.Context) error {
 	mux := http.NewServeMux()
 	mux.Handle("/environments/", http.HandlerFunc(func(w http.ResponseWriter, req *http.Request) {
 		defer readAllAndClose(req.Body, 1024)
+		if c.DexEnable {
+			role, err := auth.VerifyToken(ctx, req, c.DexClientId, c.DexBaseURL)
+			if err != nil {
+				// If user is not authenticated redirect to the login page.
+				http.Redirect(w, req, auth.LoginPATH, http.StatusFound)
+			}
+			ctx = auth.WriteUserRoleToGrpcContext(ctx, role)
+		}
 		httpHandler.Handle(w, req)
 	}))
 	mux.Handle("/release", http.HandlerFunc(func(w http.ResponseWriter, req *http.Request) {
 		defer readAllAndClose(req.Body, 1024)
+		if c.DexEnable {
+			role, err := auth.VerifyToken(ctx, req, c.DexClientId, c.DexBaseURL)
+			if err != nil {
+				// If user is not authenticated redirect to the login page.
+				http.Redirect(w, req, auth.LoginPATH, http.StatusFound)
+			}
+			ctx = auth.WriteUserRoleToGrpcContext(ctx, role)
+		}
 		httpHandler.Handle(w, req)
 	}))
```

Comment:
```suggestion
	mux.Handle("/release", http.HandlerFunc(func(w http.ResponseWriter, req *http.Request) {
		defer readAllAndClose(req.Body, 1024)
		if c.DexEnable {
			reqCtx := req.Context()
			role, err := auth.VerifyToken(reqCtx, req, c.DexClientId, c.DexBaseURL)
			if err != nil {
				// If user is not authenticated redirect to the login page.
				http.Redirect(w, req, auth.LoginPATH, http.StatusFound)
			}
			reqCtx = auth.WriteUserRoleToGrpcContext(reqCtx, role)
			req2 := req.WithContext(reqCtx)
		}
		httpHandler.Handle(w, req2)
	}))
```

Reply:
please test locally, that the context actually reaches the cd-service endpoint.
```
reqCtx = auth.WriteUserRoleToGrpcContext(reqCtx, ROLE.ENGINEER) // hardcode

// in cd service grpc endpoint:
role := auth.ReadUserRoleToGrpcContext(reqCtx) 
println(role)

```

Reply:
Did the testing. Was able to test it for the HTTP context but not for the GRPC one. I left a TODO for Gracia for that.

Reply:
Tested it for grpc!